py/objstr.c

```diff
@@ -161,6 +187,7 @@ mp_obj_t mp_obj_str_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_
                 if (str_hash == 0) {
                     str_hash = qstr_compute_hash(str_data, str_len);
                 }
+                if(!utf8_chk((uint8_t *)str_data, str_len)) mp_raise_ValueError("Not a valid UTF-8 string");
```

Comment:
Since the UnicodeError exception object is provided by uPy it would be good to use that here, like `mp_raise_msg(&mp_type_UnicodeError, "invalid utf8 string")`.  Also please not coding style, and have a single suite with braces.